File: LeetCode/Medium/2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        if (head.next == null) return null;
        
        ListNode slowest = head;
        ListNode slow = head;
        ListNode fast = head;
        
        while (fast != null && fast.next != null){
            slowest = slow;
            slow = slow.next;
            fast = fast.next.next;
        }
        
        slowest.next = slow.next;
        
        return head;
    }
}

File: LeetCode/Medium/2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteMiddle(head *ListNode) *ListNode {
    if head.Next == nil { return nil }
    
    slowest := head
    slow := head
    fast := head
    
    for fast != nil && fast.Next != nil {
        slowest = slow
        slow = slow.Next
        fast = fast.Next.Next
    }
    
    slowest.Next = slow.Next
    
    return head
}